Shopipy.UserManagement/Services/UserService.cs

```diff
@@ -0,0 +1,108 @@
+using Microsoft.AspNetCore.Identity;
+using Shopipy.Persistence.Models;
+using Shopipy.Shared.Services;
+
+namespace Shopipy.UserManagement.Services;
+
+public class UserService(UserManager<User> userManager, CurrentUserService currentUserService, IBusinessService businessService)
+{
+    public async Task CreateUserAsync(User user, string password)
+    {
+        await ValidateUserAsync(user);
+        var result = await userManager.CreateAsync(user, password);
+        if (!result.Succeeded)
+        {
+            throw new ArgumentException(result.ToString());
+        }
+    }
+
+    public async Task<IEnumerable<User>> GetAllUsersAsync()
+    {
+        var currentUser = await currentUserService.GetCurrentUserAsync();
+        if (currentUser.Role == UserRole.SuperAdmin)
+        {
+            return userManager.Users;
+        }
+
+        return userManager.Users.Where(user =>
+            user.BusinessId == currentUser.BusinessId
+        );
+    }
+
+    public async Task<User?> GetUserByIdAsync(string userId)
+    {
+        var user = await userManager.FindByIdAsync(userId);
+        if (user == null) return null;
+        try
+        {
+            await ValidateAccessToUser(user);
+        } catch (UnauthorizedAccessException)
+        {
+            return null;
+        }
+        return user;
+    }
+
+    public async Task UpdateUserAsync(User user)
+    {
+        await ValidateUserAsync(user);
+        var result = await userManager.UpdateAsync(user);
+        if (!result.Succeeded)
+        {
+            throw new ArgumentException(result.ToString());
+        }
+    }
+    
+    /// <returns>
+    /// The <see cref="Task"/> that represents the asynchronous operation, containing the boolean that is true if user was deleted.
+    /// </returns>
+    public async Task<bool> DeleteUserAsync(string userId)
+    {
+        var user = await userManager.FindByIdAsync(userId);
+        if (user == null) return false;
+        try
+        {
+            await ValidateAccessToUser(user);
+        }
+        catch (UnauthorizedAccessException)
+        {
+            return false;
+        }
+        var result = await userManager.DeleteAsync(user);
+        if (!result.Succeeded)
+        {
+            throw new ArgumentException(result.ToString());
+        }
+        return true;
+    }
+    
+    private async Task ValidateUserAsync(User user)
```

Comment:
I think it should be clear what it validates because of exception messages. Not sure regarding *how*, as it only does straightorward role/id checks. Maybe some other reviewer can take a look.